sdk/servicebus/azure-servicebus/azure/servicebus/_common/auto_lock_renewer.py

```diff
@@ -13,19 +13,26 @@
 
 from .._servicebus_session import ServiceBusSession
 from ..exceptions import AutoLockRenewFailed, AutoLockRenewTimeout, ServiceBusError
-from .utils import renewable_start_time, utc_now
+from .utils import get_renewable_start_time, utc_now, get_renewable_lock_duration
 
 if TYPE_CHECKING:
     from typing import Callable, Union, Optional
     from .message import ServiceBusReceivedMessage
-    LockRenewFailureCallback = Callable[[Union[ServiceBusSession, ServiceBusReceivedMessage],
+    Renewable = Union[ServiceBusSession, ServiceBusReceivedMessage]
+    LockRenewFailureCallback = Callable[[Renewable,
                                          Optional[Exception]], None]
 
 _log = logging.getLogger(__name__)
 
 class AutoLockRenewer(object):
     """Auto renew locks for messages and sessions using a background thread pool.
 
+    :param max_lock_renewal_duration: A time in seconds that locks registered to this renewer
+     should be maintained for. Default value is 300 (5 minutes).
+    :type max_lock_renewal_duration: float
+    :param on_lock_renew_failure: A callback may be specified to be called when the lock is lost on the renewable
+     that is being registered. Default value is None (no callback).
+    :type on_lock_renew_failure: Optional[LockRenewFailureCallback]
     :param executor: A user-specified thread pool. This cannot be combined with
      setting `max_workers`.
     :type executor: ~concurrent.futures.ThreadPoolExecutor
```

Comment:
I remember you mentioned you want to put all of them into kwargs?

Reply:
I must admit to not have a clear memory of that discussion.  I remember we were thinking about kwargification if we persisted things like renew_period, but not for these.  Do you recall the context/my reasoning?

Reply:
(from OOB, we agreed that max_worker_count might fall under this, but felt that it didn't merit the pitfalls that come from adding kwargs just to deemphesize that further than putting it at the bottom as we have.)